docs/README-maintainers.md

```diff
@@ -1,6 +1,21 @@
 README-maintainers.md
 ========
 
+
+Conventions for Git and Pull Requests
+-------------
+This repository aims to maintain a mostly linear history. In order to achieve this, please
+observe the following workflow:
+* Checkout a feature branch and open a PR when the changes are ready for review
+* After your PR has been approved, make sure to rebase your feature branch with `origin/main`.
+(Please only rebase as the last step before the merge).
```

Comment:
I would soften this statement a bit, because there are some uncommon but legitimate reasons to rebase earlier in a PR's lifetime. In particular, if the main branch has advanced in a way that potentially affects the correctness of the PR, then rebasing to test against those changes and subsequently adjust the PR accordingly may be indicated.

```suggestion
* In general you should try to avoid rebasing a non-draft PR with pending approvals until the last step before merge, because it complicates iterative review. 
```